src/tests/system/tests/test_access_control_simple.py

```diff
@@ -0,0 +1,152 @@
+"""
+SSSD Authentication Test Cases
+
+:requirement: access control
+"""
+
+from __future__ import annotations
+
+import pytest
+from sssd_test_framework.roles.client import Client
+from sssd_test_framework.roles.generic import GenericProvider
+from sssd_test_framework.topology import KnownTopologyGroup
+
+
+@pytest.mark.topology(KnownTopologyGroup.AnyProvider)
+@pytest.mark.importance("critical")
+def test_access_control_simple__permits_user_login(client: Client, provider: GenericProvider):
+    """
+    :title: Simple access filter permits user login
+    :setup:
+        1. Create users ‘user1’ and ‘user2’
+        2. Configure SSSD with ‘access_provider = simple’, ‘simple_allow_users = user1’
+        3. Start SSSD
+    :steps:
+        1. Try to login as ‘user1’
+        2. Try to login as ‘user2’
+    :expectedresults:
+        1. User1 can login
+        2. User2 cannot login
+    :customerscenario: False
+    """
+
+    provider.user("user1").add()
+    provider.user("user2").add()
+
+    client.sssd.domain["access_provider"] = "simple"
+    client.sssd.domain["simple_allow_users"] = "user1"
+
+    client.sssd.start()
+
+    assert client.auth.ssh.password("user1", "Secret123"), "User should be able to log in!"
+    assert not client.auth.ssh.password("user2", "Secret123"), "User should NOT be able to log in!"
+
+
+@pytest.mark.topology(KnownTopologyGroup.AnyProvider)
+@pytest.mark.importance("critical")
+def test_access_control_simple__deny_user_login(client: Client, provider: GenericProvider):
+    """
+    :title: Simple access filter permits user login
+    :setup:
+        1. Create users ‘user1’ and ‘user2’
+        2. Configure SSSD with ‘access_provider = simple’, ‘simple_deny_users = user1’
+        3. Start SSSD
+    :steps:
+        1. Try to login as ‘user1’
+        2. Try to login as ‘user2’
+    :expectedresults:
+        1. User1 cannot login
+        2. User2 can login
+    :customerscenario: False
+    """
+    provider.user("user1").add()
+    provider.user("user2").add()
+
+    client.sssd.domain["access_provider"] = "simple"
+    client.sssd.domain["simple_deny_users"] = "user1"
+
+    client.sssd.start()
+
+    assert not client.auth.ssh.password("user1", "Secret123"), "User should be able to log in!"
+    assert client.auth.ssh.password("user2", "Secret123"), "User should NOT be able to log in!"
+
+
+@pytest.mark.topology(KnownTopologyGroup.AnyProvider)
+@pytest.mark.importance("critical")
+def test_access_control_simple__permits_user_login_based_on_group(client: Client, provider: GenericProvider):
+    """
+    :title: Simple access filter permits user login
```

Comment:
Simple access filter permits user based on groups  OR

Simple access filter permits and denies access evaluating groups